src/Server/Utils/Strategies/RegexResponseStrategy.php

```diff
@@ -67,13 +75,25 @@ private function bodyConditionIsRegex($expectation)
     private function fillWithUrlMatches($expectation, $httpRequest, $responseBody)
     {
         if ($this->urlConditionIsRegex($expectation)) {
-            $responseBody = preg_replace('/\$\{url\.(\d+)\}/', '\$$1', $responseBody);
-
-            return preg_replace(
-                $expectation->getRequest()->getUrl()->getValue(),
-                $responseBody,
-                $httpRequest->getUri()->__toString()
+            $pattern = preg_replace(
+                '/^(.)\^/',
```

Comment:
Just a question. Why is this replacement needed?

Reply:
It's needed for strict regexp for URL, when defined "~^some/url$~" it sussefuly routes for "http://host:port/some/url", but the full url is returned from `$httpRequest->getUri()->__toString()`, and "http://host:port/some/url"  doesn't matches "~^some/url$~" here.
So I remove "^" symbol from initial expression to get matches, saving expression open symbol.

Reply:
I would prefer not to modify the regex created in the expectation. What do you think of using parse_url to get only the path and query string from the url so we can use them instead of $httpRequest->getUri()->__toString()?

Reply:
Now I see that we can get that data directly from Uri object. So no parse_url needed.